src/core/algorithms/association_rules/ar_verifier/ar_verifier.cpp

```diff
@@ -0,0 +1,112 @@
+#include "ar_verifier.h"
+
+#include <chrono>
+#include <stdexcept>
+
+#include <easylogging++.h>
+
+#include "config/equal_nulls/option.h"
+#include "config/indices/option.h"
+#include "config/names_and_descriptions.h"
+#include "config/option_using.h"
+#include "config/tabular_data/input_table/option.h"
+#include "util/timed_invoke.h"
+
+namespace algos::ar_verifier {
+ARVerifier::ARVerifier() : Algorithm({}) {
+    RegisterOptions();
+    using namespace config::names;
+    MakeOptionsAvailable({kTable, kInputFormat});
+}
+
+void ARVerifier::RegisterOptions() {
+    DESBORDANTE_OPTION_USING;
+
+    auto sing_eq = [](InputFormat input_format) { return input_format == +InputFormat::singular; };
+    auto tab_eq = [](InputFormat input_format) { return input_format == +InputFormat::tabular; };
+
+    RegisterOption(config::kTableOpt(&input_table_));
+    RegisterOption(Option{&first_column_tid_, kFirstColumnTId, kDFirstColumnTId, false});
+    RegisterOption(Option{&item_column_index_, kItemColumnIndex, kDItemColumnIndex, 1u});
+    RegisterOption(Option{&string_rule_left_, kARuleLeft, kDARuleLeft, std::list<std::string>{}});
+    RegisterOption(
+            Option{&string_rule_right_, kARuleRight, kDARuleRight, std::list<std::string>{}});
+    RegisterOption(Option{&minconf_, kMinimumConfidence, kDMinimumConfidence, 0.0});
+    RegisterOption(Option{&minsup_, kMinimumSupport, kDMinimumSupport, 0.0});
+    RegisterOption(Option{&tid_column_index_, kTIdColumnIndex, kDTIdColumnIndex, 0u});
+    RegisterOption(Option{&input_format_, kInputFormat, kDInputFormat}.SetConditionalOpts(
+            {{sing_eq, {kTIdColumnIndex, kItemColumnIndex}}, {tab_eq, {kFirstColumnTId}}}));
+}
+
+void ARVerifier::MakeExecuteOptsAvailable() {
+    using namespace config::names;
+    MakeOptionsAvailable({kARuleLeft, kARuleRight, kMinimumSupport, kMinimumConfidence});
+}
+
+void ARVerifier::LoadDataInternal() {
+    switch (input_format_) {
+        case InputFormat::singular:
+            transactional_data_ = model::TransactionalData::CreateFromSingular(
+                    *input_table_, tid_column_index_, item_column_index_);
+            break;
+        case InputFormat::tabular:
+            transactional_data_ =
+                    model::TransactionalData::CreateFromTabular(*input_table_, first_column_tid_);
+            break;
+        default:
+            assert(0);
+    }
+    if (transactional_data_->GetNumTransactions() == 0) {
+        throw std::runtime_error("Got an empty dataset: AR verifying is meaningless.");
+    }
+}
+
+unsigned long long ARVerifier::ExecuteInternal() {
+    std::vector<std::string> const& item_names_map = transactional_data_->GetItemUniverse();
+
+    std::vector<unsigned> ar_left_id;
```

Comment:
It seems more correct to use `model::ColumnIndex`. If this is true, then it should be here and in those places where we refer to the index inside the rule.